### PR TITLE
Before after gifs

### PR DESCRIPTION
**Pull request by guidotheelen**

### CODE REVIEW

File: Lib/diffenator/__init__.py [hunk 1]
```diff
@@ -293,10 +291,38 @@ def __init__(self, table_name, font_a, font_b,
         self._font_b = font_b
 
     def to_gif(self, dst, padding_characters="", limit=800):
-
         if not self._font_a.size or not self._font_b.size:
-            logger.info("Font can't be resized, skipping before and after gifs")
-            return
+            if "cbdt" in dst:
+                dstDir = dst.replace("cbdt_glyphs_modified.gif", "")
```
Comment (RoelN) [inline Lib/diffenator/__init__.py:296]: Is this being used?

File: Lib/diffenator/__init__.py [hunk 2]
```diff
@@ -293,10 +291,38 @@ def __init__(self, table_name, font_a, font_b,
...
+                for element in self._data:
+                    key_before = element["glyph before"]
+                    key_after = element["glyph after"]
+
+                    with open(f'{dstDir}{key_before}_old.png', 'wb') as f:
+                        image1 = font_a_images[key_before].convert("RGBA")
+                        font_a_images[key_before].save(f, "PNG")
+                    with open(f'{dstDir}{key_after}_new.png', 'wb') as f:
+                        image2 = font_b_images[key_after].convert("RGBA")
+                        image2.save(f, "PNG")
```
[the remark this answers is not in the document]
Comment (guidotheelen) [reply Lib/diffenator/__init__.py:310]: I'll try to fix it in memory, would be nicer! 🤔

File: Lib/diffenator/__init__.py [hunk 3]
```diff
@@ -293,10 +291,38 @@ def __init__(self, table_name, font_a, font_b,
...
+
+                font_a_images = read_cbdt(self._font_a.ttfont)
+                font_b_images = read_cbdt(self._font_b.ttfont)
+
+                for element in self._data:
+                    key_before = element["glyph before"]
+                    key_after = element["glyph after"]
+
+                    with open(f'{dstDir}{key_before}_old.png', 'wb') as f:
+                        image1 = font_a_images[key_before].convert("RGBA")
```
Comment (RoelN) [inline Lib/diffenator/__init__.py:306]: These are already RGBA?

File: Lib/diffenator/__init__.py [hunk 4]
```diff
@@ -293,10 +291,38 @@ def __init__(self, table_name, font_a, font_b,
...
+                        font_a_images[key_before].save(f, "PNG")
+                    with open(f'{dstDir}{key_after}_new.png', 'wb') as f:
+                        image2 = font_b_images[key_after].convert("RGBA")
```
Comment (RoelN) [inline Lib/diffenator/__init__.py:309]: These are already RGBA?

File: Lib/diffenator/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (guidotheelen) [reply Lib/diffenator/__init__.py]: Done ✅

File: Lib/diffenator/__init__.py (hunk 3 above)
Comment (guidotheelen) [reply Lib/diffenator/__init__.py]: Removed it!

File: Lib/diffenator/__init__.py (hunk 4 above)
Comment (guidotheelen) [reply Lib/diffenator/__init__.py]: Removed it!

File: Lib/diffenator/__init__.py (hunk 1 above)
Comment (guidotheelen) [reply Lib/diffenator/__init__.py:296]: Yeah dst is always: `outputdir/cbdt_glyphs_modified.gif`. I Only need `outputdir/`

File: Lib/diffenator/__init__.py [hunk 5]
```diff
@@ -293,10 +291,38 @@ def __init__(self, table_name, font_a, font_b,
...
+                            duration=1000,
+                            loop=10000)
+                    os.remove(f"{dstDir}{key_before}_old.png")
+                    os.remove(f"{dstDir}{key_after}_new.png")
+
+                logger.info(f"images are stored in: {dstDir}")
+                return
+
+            else:
+                logger.info(f"Font can't be resized, skipping {dst.split('/')[-1]}")
```
Comment (RoelN) [inline Lib/diffenator/__init__.py:315]: What's being logged here? Is this the intended output:

```
INFO:fontdiffenator:Font can't be resized, skipping glyphs_missing.gif
INFO:fontdiffenator:Font can't be resized, skipping metrics_modified.gif
```

File: Lib/diffenator/__init__.py [hunk 6]
```diff
@@ -293,10 +291,38 @@ def __init__(self, table_name, font_a, font_b,
         self._font_b = font_b
 
     def to_gif(self, dst, padding_characters="", limit=800):
-
         if not self._font_a.size or not self._font_b.size:
-            logger.info("Font can't be resized, skipping before and after gifs")
-            return
+            if "cbdt" in dst:
```
[the remark this answers is not in the document]
Comment (guidotheelen) [reply Lib/diffenator/__init__.py]: Done 👍

**Pull request merged**